draft-ietf-httpbis-header-structure.md

```diff
@@ -211,32 +203,64 @@ In HTTP headers, each member is separated by a comma and optional whitespace. Fo
 Example-StrListHeader: "foo", "bar", "It was the best of times."
 ~~~
 
-In HTTP headers, inner lists are denoted by surrounding parenthesis, and have their values delimited by a single space. A header field whose value is defined as a list of lists of strings could look like:
+In HTTP headers, an empty list is denoted by not serialising the header at all.
+
+Parsers MUST support lists containing at least 1024 members. Header specifications can constrain the types and cardinality of individual list values as they require.
+
+
+### Inner Lists {#inner-list}
+
+An inner list is an array of zero or more items ({{item}}). Both the individual items and the inner-list itself can be parameterised ({{param}}).
+
+The ABNF for inner-lists in HTTP headers is:
+
+~~~ abnf
+inner-list    = "(" OWS [ sh-item *( SP sh-item ) OWS ] ")" *parameter
```

Comment:
Not related to this PR per se, but I just realised the ABNF for inner list says exactly `SP` between items, but the parsing algorithm accepts `SP` followed by any other `OWS`

Reply:
Hm. IIRC my thinking when I wrote this was that the ABNF was illustrating the "correct" on-wire form, rather than being a way to write a parser. Still, we could import `RWS` and use that...

Reply:
I've absolutely no doubt someone will assume the ABNF and the algorithms match, and write a regexp parser based on the former.

(_sotto voce_) `RWS` still doesn't work, because it must start with `SP`

Reply:
So maybe we should make the parsing algorithm more strict, either `SP` or `1*SP`

Reply:
Just `SP` works for me. That way it couldn't mask something like a null value that gets written as a zero-length string, for example.